src/Controller/Api/FileController.php

```diff
@@ -51,10 +55,88 @@ public function deleteFile(File $file, MessageBusInterface $messageBus, EntityMa
 
         return new Response(
             null,
-            Response::HTTP_NO_CONTENT,
+            Response::HTTP_OK,
             array(
                 'Content-Type' => 'application/x-empty',
             )
         );
     }
+
+    /**
+     * Update a file entity.
+     *
+     * @param File                   $file          File entity instance.
+     * @param EntityManagerInterface $entityManager Entity manager interface instance.
+     * @param Request                $request       The request body sent with destination directory.
+     * @param Datastore              $datastore     Datastore to manipulate the file on disk.
+     *
+     * @Route("/api/file/{id}", name="pelagos_api_file_update_filename", methods={"PUT"}, defaults={"_format"="json"})
+     *
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     * @IsGranted("CAN_EDIT", subject="file")
+     *
+     * @throws BadRequestHttpException When the destination file name already exists.
+     *
+     * @return Response A response object with an empty body and a "no content" status code.
+     */
+    public function updateFileName(File $file, EntityManagerInterface $entityManager, Request $request, Datastore $datastore) : Response
+    {
+        $newFileName = $request->get('destinationDir');
+        $fileset = $file->getFileset();
+        if (!$fileset->doesFileExist($newFileName)) {
+            // Rename file on disk if it is already processed
+            if ($file->getStatus() === File::FILE_DONE) {
+                $datastore->renameFile($file->getFilePathName(), $newFileName);
+            }
+            $file->setFilePathName($newFileName);
+            $entityManager->flush();
+        } else {
+            throw new BadRequestHttpException('File with same name and folder already exists');
+        }
+
+        return new Response(
+            null,
+            Response::HTTP_OK,
+            array(
+                'Content-Type' => 'application/x-empty',
+            )
+        );
+    }
+
+    /**
+     * Download a file from disk.
+     *
+     * @param File                   $file          File entity instance.
+     * @param Datastore              $datastore     Datastore to manipulate the file on disk.
+     *
+     * @Route("/api/file/download/{id}", name="pelagos_api_file_download", defaults={"_format"="json"})
+     *
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     * @IsGranted("CAN_EDIT", subject="file")
```

Comment:
as soon as the voter is ready, we'll have to make such that the user has to be in the same RG as the file, but this is fine for now imho.